### PR TITLE
Check Homebrew for common libraries on macOS

### PR DESCRIPTION
**Pull request by pythonengineer**
I don't know if this is the most optimal way of including the path, but some Mac users will have the libraries Pyglet can depend upon (like for audio) installed through Homebrew and they may not be caught through the existing library paths that are checked by default.

### CODE REVIEW

File: pyglet/lib.py [hunk 1]
```diff
@@ -164,6 +164,9 @@ def __init__(self):
         else:
             self.dyld_fallback_library_path = [os.path.expanduser('~/lib'), '/usr/local/lib', '/usr/lib']
 
+        # check homebrew for libs too
+        self.brew_library_path = ['/opt/homebrew/lib']
```
Comment (pushfoo) [inline pyglet/lib.py:168]: This might be better as a keyword argument. The doc seems to strongly recommend against changing the directory, [but it does seem to support it in theory](https://docs.brew.sh/Installation#untar-anywhere-unsupported) if I understood it correctly.

File: pyglet/lib.py [hunk 2]
```diff
@@ -205,11 +208,13 @@ def find_library(self, path):
             search_path.extend([os.path.join(p, libname) for p in self.dyld_library_path])
             search_path.append(path)
             search_path.extend([os.path.join(p, libname) for p in self.dyld_fallback_library_path])
+            search_path.extend([os.path.join(p, libname) for p in self.brew_library_path])
```
Comment (pushfoo) [inline pyglet/lib.py:211]: This seems like it shouldn't be an issue, but I'd double check to make sure this doesn't break conda or venvs.

**Review by pushfoo — CHANGES_REQUESTED** [review]
tl;dr:
1. brew seems like it provides but discourages a way to customize install directory
2. the brew directory should probably be a keyword argument with stern warnings

**Comment by pythonengineer** [comment]
The keyword argument with warning is added.

File: pyglet/lib.py (hunk 2 above)
Comment (benmoran56) [reply pyglet/lib.py:214]: This should be fine as long as it's at the end. Other locations will already have been tried.

**Comment by benmoran56** [comment]
Closing due to lack of follow up.
